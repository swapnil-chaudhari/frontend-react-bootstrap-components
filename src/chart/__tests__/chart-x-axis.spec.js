import React from 'react';
import renderShallow from 'render-shallow';
import ChartXaxis from '../chart-x-axis';

describe('<ChartXaxis>', () => {
    describe('when it renders', () => {
        let component;
        const ticks = ['2018-03-12', '2018-03-24', '2018-04-05'];
        const props = {
            ticks,
            height: 400,
            xAxisOffset: 50,
            yAxisOffset: -25,
            width: 888,
            scale: (i) => i
        };

        beforeAll(() => {
            component = renderShallow(
                <ChartXaxis { ...props } />
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it has hover', () => {
        let component;
        const hover = {
            hoverText: 'February 15th, 2005',
            hoverPosition: 120
        };
        const ticks = ['2018-03-12', '2018-03-24', '2018-04-05'];
        const props = {
            hover,
            ticks,
            height: 400,
            xAxisOffset: 50,
            yAxisOffset: -25,
            width: 888,
            scale: (i) => i
        };

        beforeAll(() => {
            component = renderShallow(
                <ChartXaxis { ...props } />
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
