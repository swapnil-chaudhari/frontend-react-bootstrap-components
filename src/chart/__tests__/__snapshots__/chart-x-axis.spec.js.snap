// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartXaxis> when it has hover has proper markup 1`] = `
<g
  className="ChartXaxis"
  transform="translate(50, 0)"
>
  <text
    className="ChartXaxis-hover-text"
    textAnchor="middle"
    x={70}
    y={400}
  >
    February 15th, 2005
  </text>
</g>
`;

exports[`<ChartXaxis> when it renders has proper markup 1`] = `
<g
  className="ChartXaxis"
  transform="translate(50, 0)"
>
  <g>
    <text
      className="ChartXaxis-label"
      x="2018-03-12"
      y={400}
    >
      Mar 12, 2018
    </text>
  </g>
  <g>
    <text
      className="ChartXaxis-label"
      x="2018-03-24"
      y={400}
    >
      Mar 24, 2018
    </text>
  </g>
  <g>
    <text
      className="ChartXaxis-label"
      x="2018-04-05"
      y={400}
    >
      Apr 5, 2018
    </text>
  </g>
</g>
`;
